### DISCUSSION TITLE
Migrate from Firebase

### DISCUSSION THREAD

**Discussion by chrisdrobison** [zitadel/zitadel#5525] [Q&A]
Just wondering if you have any suggestions for someone who might want to migrate out of Firebase Auth (Google Identity Platform) to this platform.

**Comment by hifabienne** [comment]
Hei @chrisdrobison 
We do not have a ready process on how to migrate from firebase to ZITADEL yet.

I think there are different things to consider.
- Is downtime acceptable while you migrate from your legacy system to ZITADEL?
- Can you move all apps to ZITADEL? in one go?
- Are all the user information available via API from the legacy system? Which password hash algorithm does your system use?

In general a migration process could look something like this:
1. User logs into your “old” login in your case with firebase auth
2. In the back you create a new user in ZITADEL with the password the user already had
3. Flag the user in firebase as “migrated”
4. On the next login check if the user is flagged as “migrated”, redirect directly to ZITADEL login instead of the old login

With this process you wouldn’t have any downtime and you can migrate the users from firebase auth to ZITADEL one by one.

**Comment by fcoppede** [comment]
we added the following migration guide to our docs regarding Firebase:

- https://zitadel.com/docs/guides/migrate/sources/firebase

Thanks!
